crates/cargo-wdk/src/providers/exec.rs

```diff
@@ -27,7 +23,14 @@ use super::error::CommandError;
 #[derive(Debug, Default)]
 pub struct CommandExec {}
 
-#[automock]
+#[cfg_attr(test, mockall::automock)]
+#[cfg_attr(
+    test,
+    allow(
+        dead_code,
+        reason = "This implementation is mocked in test configuration."
```

Comment:
The reason is not clear. It does not quite explain _why_ we are allowing dead code.

Reply:
Okay. How about this?

"The implementation is mocked, so the actual implementation will appear as dead code in test configuration"

Reply:
I updated the reason to

`Tests use mocked implementation, so this implementation becomes dead code in test configuration.`

Reply:
"impl" is more precise than "implementation" in the context of Rust and shorter, But it's not a big deal.